cmb_lensing/kappa_ell_cut/kappa_ell_cut.py

```diff
@@ -21,14 +21,42 @@ def apply_lcut(block, section, Lmin, Lmax):
     w   = np.ones(ell.shape[0])
     w[ell<Lmin]=0
     w[ell>Lmax+1]=0
+
     for bini in range(0,n_other):
         C_ell_orig = block[section, 'bin_{}_1'.format(bini+1)]
-        beamed_C_ell = C_ell_orig*w
-        block[section, 'bin_{}_1'.format(bini+1)] = beamed_C_ell
+        filtered_C_ell = C_ell_orig*w
+        if (not output_section is None):
+            block[output_section, 'bin_{}_1'.format(bini+1)] = filtered_C_ell
+        if (output_section is None):
+            block[section, 'bin_{}_1'.format(bini+1)] = filtered_C_ell
+
+    if  (section != output_section and (not output_section is None)):
+        if (block.has_value(section,"nbin")):
+            block[output_section, "nbin"] = block[section, "nbin"]
+        if (block.has_value(section,"nbin_a")):
+            block[output_section, "nbin_a"] = block[section, "nbin_a"]
+        if (block.has_value(section,"nbin_b")):
+            block[output_section, "nbin_b"] = block[section, "nbin_b"]
+        block[output_section, "save_name"] =  block[section, "save_name"]
+        block[output_section, "ell"] = block[section, "ell"]
+        block[output_section, "sep_name"] = block[section, "sep_name"]
+        block[output_section, "sample_a"] = block[section, "sample_a"]
+        block[output_section, "sample_b"] = block[section, "sample_b"]
+        block[output_section, "is_auto"] = block[section, "is_auto"]
+        if (save_name is not None):
+            block[output_section, "save_name"] = save_name
 
 def setup(options):
     shearkappa_section = options.get_string(option_section, "shearkappa_section", default="shear_cmbkappa_cl")
     galkappa_section = options.get_string(option_section, "galkappa_section", default="galaxy_cmbkappa_cl")
+    kappakappa_section = options.get_string(option_section, "kappakappa_section", default="cmbkappa_cl")
+    shearkappa_output_section = options.get_string(option_section, "shearkappa_output_section", default=None)
```

Comment:
Setting the default as None for these parameters means they are required and will raise an error if they're not set. I think you were aiming to make them optional here? If so I usually set the default to just `""`.

Then you might need to change the check in the apply_lcut function